### PR TITLE
Kdump page improvements

### PR DESCRIPTION
**Pull request by skobyda**
- Remove service status, use Alert instead for situations when service is failed / not installed / kernelcrash parameter is not set
- 'Test configuration' should use danger variant button
- Show target location on the card
- Reorganize content of kdump card 
- Add title to kdump card 

### New design:
image

With SSH:
image

With NFS:
image

### kernelcrash parameter is not set
image

### Service is not installed:
image

### Service is failed
image

Designs are available here https://github.com/cockpit-project/cockpit/discussions/19203#discussion-5527715

### CODE REVIEW

File: pkg/kdump/kdump-view.jsx
Comment (cockpituous) [inline pkg/kdump/kdump-view.jsx:285]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19240-20230824-135223-f118a980-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/kdump/kdump-view.jsx
Comment (cockpituous) [inline pkg/kdump/kdump-view.jsx:296]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19240-20230824-135223-f118a980-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/kdump/kdump-view.jsx
Comment (cockpituous) [inline pkg/kdump/kdump-view.jsx:444]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19240-20230824-135223-f118a980-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/kdump/kdump-view.jsx
Comment (cockpituous) [inline pkg/kdump/kdump-view.jsx:447]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19240-20230824-135223-f118a980-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/kdump/kdump-view.jsx
Comment (cockpituous) [inline pkg/kdump/kdump-view.jsx:453]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19240-20230824-135223-f118a980-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by skobyda** [comment]
Few notes:

This PR is a subset of https://github.com/cockpit-project/cockpit/discussions/19203

In this PR, we do not yet add a history of previous crashes, as it might be way to complex to figure out. We only prepare the design, so we can later add the status feature. This feature will be able to detect if kdump configuration is up to date, failed and if test of its configuration went succesfully:
 
d6bf0579-a1ae-4780-b705-3325b7f9a11f
8afcb9f7-8997-451f-990d-e84af08fb6d2
95765854-78c4-46ad-8470-3ab10fcdeaa3
f0539452-acfe-49d1-b219-86ccc7735065

**Comment by martinpitt** [comment]
I locally rebased this. It mostly looks fine, but he location edit button is misaligned, and it should also say "Edit", not "Change", to be consistent with other Cockpit pages:

image

And of course the tests need updating.

**Comment by martinpitt** [comment]
I fixed up the tests, and updated the link button from "Change" to "Edit". I still want to add some more tests for "service is not running" and "service not available", see the coverage report above. I'm updating the screenshots now.

**Comment by martinpitt** [comment]
@garrett This is a first set of steps towards the [new design](https://github.com/cockpit-project/cockpit/discussions/19203#discussioncomment-6742834). Needs more steps obviously (like the list of recent crashes), but it'd be nice to finish Simon's old PR first and land the next steps as follow-ups?

File: pkg/kdump/kdump-view.jsx
Comment (garrett) [inline pkg/kdump/kdump-view.jsx:549]:
```suggestion
                    alertMessage = _('Kernel did not boot with the "crashkernel" setting');
```

File: pkg/kdump/kdump-view.jsx
Comment (garrett) [inline pkg/kdump/kdump-view.jsx:560]:
```suggestion
                alertMessage = _("Kdump service is not installed");
```

File: pkg/kdump/kdump-view.jsx
Comment (garrett) [inline pkg/kdump/kdump-view.jsx:561]:
```suggestion
                alertDetail = _("Install the `kexec-tools` package.");
```

File: pkg/kdump/kdump-view.jsx
Comment (garrett) [inline pkg/kdump/kdump-view.jsx:550]:
```suggestion
                    alertDetail = _("Reserve memory at boot time by setting a `crashkernel` option on the kernel command line. For example, append `crashkernel=512M` to `GRUB_CMDLINE_LINUX` in `/etc/default/grub` or use your distribution's kernel argument editor.");
```

File: pkg/kdump/kdump.scss
Comment (garrett) [inline pkg/kdump/kdump.scss:38]: We could use the utility class `pf-v5-u-ml-sm` instead, now that PatternFly supports RTL.

That helper class adds this, according to Firefox's devtools:

```css
.pf-v5-u-ml-sm {
  -webkit-margin-start: var(--pf-v5-global--spacer--sm) !important;
  margin-inline-start: var(--pf-v5-global--spacer--sm) !important;
}
```

(`-webkit-margin-start` is an unneeded alias that they're using for whatever reason. The intent is probably to support even older browsers... but Safari added `margin-inline-start` in 2019, Chrome in 2018, Edge in 2020 (when Edge switched to Chromium), and Firefox way back in 2015. So it's really very well supported and not needed. At least by us, and probably really not by PatternFly either.)

**Review by garrett — CHANGES_REQUESTED** [review]
We should merge this in ASAP, as-is and do follow-ups, as it's already better than what we have and is headed in the right direction, but still needs some work to match the mockups.

However, I've added some string change suggestions and have also pointed out a place where custom CSS could be replaced by a PatternFly helper class. If we could take care of this all first, then I think we can merge it after the changes and then do the follow-ups.

BTW: I'm using backticks in the strings... it would be ideal if we could use the Markdown renderer for these to have them show up as pre blocks. (We've done this in other places for alerts, modals titles, and such, for very minimal Markdown formatting.)

File: pkg/kdump/kdump-view.jsx
Comment (garrett) [reply pkg/kdump/kdump-view.jsx:549]: This could also be backticks.

```suggestion
                    alertMessage = _("Kernel did not boot with the `crashkernel` setting");
```

However, I'm not sure how it would look with a `<pre>` in the title.

File: pkg/kdump/kdump.scss
Comment (martinpitt) [reply pkg/kdump/kdump.scss:38]: That works well indeed! It gets broken (no spacing at all) with dropping the custom class, and looks fine again with the pf helper class. Done.

**Comment by martinpitt** [comment]
@garrett Thanks for your suggestions! I applied them all, abeit not with Markdown (overkill) but with the much simpler `fmt_to_fragments()`, we already use this approach in many other places. I updated the screenshots in the description.

**Comment by martinpitt** [comment]
@jelly As I messed around this PR quite a bit, I'd appreciate another pair of eyes. This should wait for @garrett 's final sign-off, but as he already reviewed and I applied all of his suggestions, I'm 90% certain it'll be fine for him.

**Review by jelly — APPROVED** [review]

**Comment by jelly** [comment]
I'm still a bit on the fence about the alert when kdump does not start, it is not something I believe we do consistently in all our pages.

**Review by garrett — APPROVED** [review]
I think this is good enough for a first pass. We still need follow-ups, of course. :wink: 

Thanks, @skobyda and @martinpitt for this PR! :clap: :partying_face:

**Pull request merged**
